app/workers/publishing_api_vanish_worker.rb

```diff
@@ -0,0 +1,12 @@
+class PublishingApiVanishWorker < PublishingApiWorker
+  def perform(content_id, locale)
+    Services.publishing_api.unpublish(
+      content_id,
+      type: "vanish",
```

Comment:
From what I remember about the intention behind supporting this as an unpublishing type, it was about removing sensitive content from the site.

From the commit message, I gather that this is about getting a 404, rather than a 410, but whatever content has been published was still published. Is there something odd about the content that is being unpublished here?

Reply:
To clarify, this makes the "coming soon" page a 404 because the content that was coming soon was never published. So the content itself isn't there, and it doesn't make sense for the page that was announcing some upcoming content to be replaced with a 410.